### PR TITLE
make allowed_update type safe

### PR DESCRIPTION
**Pull request by johannesvollmer**
instead of having string values, use json to generate strings from a fixed set of enum values

### CODE REVIEW

File: src/objects.rs [hunk 1]
```diff
@@ -330,7 +330,29 @@ pub struct WebhookInfo {
...
     #[serde(skip_serializing_if = "Option::is_none")]
     #[builder(setter(into, strip_option), default)]
-    pub allowed_updates: Option<Vec<String>>,
+    pub allowed_updates: Option<Vec<AllowedUpdate>>,
+}
+
+/// Control which updates to receive.
+/// Specify an empty list to receive all update types except `ChatMember`.
+/// [Official documentation](https://core.telegram.org/bots/api#getupdates).
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
```
Comment (EdJoPaTo) [inline src/objects.rs:339]: No need for references with `Copy`
```suggestion
#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
```

File: src/objects.rs [hunk 2]
```diff
@@ -330,7 +330,29 @@ pub struct WebhookInfo {
...
+#[serde(rename_all = "snake_case")]
+pub enum AllowedUpdate {
```
Comment (EdJoPaTo) [inline src/objects.rs:341]: Maybe mark this enum as `#[non_exhaustive]`?

(Maybe that might be good for some more enums in this crate?

**Review by EdJoPaTo — CHANGES_REQUESTED** [review]
I think the examples should not contain the allowed updates. The examples are meant to be simple, not to confuse with additional stuff to think about.

I would remove them from the examples.

**Comment by ayrat555** [comment]
@johannesvollmer your suggestion makes sense. and the pr looks good to me

when @EdJoPaTo approves we can merge

**Comment by EdJoPaTo** [comment]
> We can simplify the examples like this

Looks good. Lets get rid of the `allowed_updates` in the examples.

**Comment by johannesvollmer** [comment]
Although, I would recommend to add separate more sophisticated examples later. Especially for people understanding Rust, but with less experience in using the Telegram API. For me, after reading the example, I understood what the `GetUpdatsParams.offset` field is supposed to do

**Review by EdJoPaTo — APPROVED** [review]

**Comment by ayrat555** [comment]
Thank you for your work, guys. We should move the library to the organisation (so it won't be associated only with me). But it seems it's overhead to create an organisation for a simple library. 

At least I need to remove my sponsor links from the library

**Comment by johannesvollmer** [comment]
You two could do it, but for me it's fine as-is :)

**Comment by EdJoPaTo** [comment]
Nah, keep it as it is. No need for organizations for everything. You can also keep the sponsoring stuff if you want to. I am fine with one person in there while others are helping.
(Which reminds me: I should continue setting up my sponsor stuff sometime…)

**Pull request merged**